src/python/pants/option/global_options.py

```diff
@@ -874,11 +874,14 @@ def register_options(cls, register):
             "tags ('-' prefix).  Useful with ::, to find subsets of targets "
             "(e.g., integration tests.)",
         )
+        ci_var = os.environ.get("CI")
         register(
             "--dynamic-ui",
             type=bool,
-            default=sys.stderr.isatty(),
-            help="Display a dynamically-updating console UI as pants runs.",
+            default=(ci_var != "true"),
```

Comment:
How about keeping the `isatty()` check too? `default="CI" in os.environ and sys.stderr.isatty()`

I'm concerned this would now break things like SSHing into a server and running Pants.

Reply:
Yea, still need to check for the TTY.